code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+	///mob the hologram will copy
+	var/mob/living/holo_host
+	///used to remember the recent speech of the holo_host
+	var/list/recent_speech = list()
+
+
+/obj/effect/warped_rune/ceruleanspace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/ceruleanspace/process()
+	if(cooldown > world.time)
+		return
+	cooldown = max_cooldown + world.time
+	if(holotile && length(recent_speech))
+		holotile.say(recent_speech[pick(recent_speech)]) //say one of the 5 latest sentence said by the holo_host
+
+
+///makes a hologram of the mob stepping on the tile, any new person stepping in will replace it with a new hologram
+/obj/effect/warped_rune/ceruleanspace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(!istype(crossing,/mob/living))
+		return
+
+	if(locate(holotile) in rune_turf)//here to delete the previous hologram,
+		qdel(holotile)
+
+	holo_host = crossing
+	holotile = new(rune_turf) //setting up the hologram to look like the person that just stepped in
+	holotile.icon = holo_host.icon
+	holotile.icon_state = holo_host.icon_state
+	holotile.alpha = 100
+	holotile.name = "[holo_host.name] (Hologram)"
+	holotile.add_atom_colour("#77abff", FIXED_COLOUR_PRIORITY)
+	holotile.copy_overlays(holo_host, TRUE)
+
+
+//the code that follows is basically the code that changeling use to get people's last spoken sentences with a few tweaks.
+	recent_speech = list() //resets the list from its previous sentences
+	var/list/say_log = list()
+	var/log_source = holo_host.logging
+	for(var/log_type in log_source)
+		var/nlog_type = text2num(log_type)
+		if(nlog_type & LOG_SAY)
+			var/list/reversed = log_source[log_type]  //reverse the list so we get the last sentences instead of the first
+			if(islist(reversed))
+				say_log = reverseRange(reversed.Copy())
+				break
+
+	if(length(say_log) > 10) //we're going to get up to the last 10 sentences spoken by the holo_host
+		recent_speech = say_log.Copy(say_log.len - 11,0)
+	else
+		for(var/spoken_memory in say_log)
+			if(recent_speech.len >= 10)
+				break
+			recent_speech[spoken_memory] = say_log[spoken_memory]
+
+
+///destroys the hologram with the rune
+/obj/effect/warped_rune/ceruleanspace/Destroy()
+	qdel(holotile)
+	holotile = null
+	holo_host = null
+	recent_speech = null
+	return ..()
+
+
+/obj/item/slimecross/warping/pyrite
+	colour = "pyrite"
+	runepath = /obj/effect/warped_rune/pyritespace
+	effect_desc = "draws a rune that will randomly color whoever steps on it"
+
+
+/obj/effect/warped_rune/pyritespace
+	desc = "Who shall we be today? they asked, but not even the canvas would answer."
+	icon_state = "colorune"
+
+
+///colors whatever steps on the rune randomly
+/obj/effect/warped_rune/pyritespace/Crossed(atom/movable/AM)
+	. = ..()
+	AM.color = rgb(rand(0,255),rand(0,255),rand(0,255))
+
+
+/* Will make anyone on the rune do genetic punch damage. May need rebalancing. Not compatible with gloves of the north star*/
+
+
+/obj/item/slimecross/warping/red
```

Comment:
awful, AWFUL rune

literally bottom tier

not useful for griefing people, killing people, OR helping people

literally when would you ever use this

Reply:
if you REALLY want to stick with the "soul-punching fists" idea, AT LEAST make the rune make you immovable while you're alive and on it

Reply:
why not try something like the blood beam ability of the blood rites cult spell? that ability's cool, but it's so impractical that we barely ever see it (but hey, it's more practical than this rune in its current state)

Reply:
or maybe just cut the soul and fist stuff and just add an effect that gives all of your attacks (including your ranged attacks) lifesteal